app/components/CollectionPreview.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+import * as React from "react";
+import { Link } from "react-router-dom";
+import styled, { withTheme } from "styled-components";
+import Collection from "models/Collection";
+import CollectionIcon from "components/CollectionIcon";
+import Flex from "components/Flex";
+import Highlight from "components/Highlight";
+import CollectionMenu from "menus/CollectionMenu";
+
+type Props = {
+  collection: Collection,
+  highlight?: ?string,
+};
+
+class CollectionPreview extends React.Component<Props> {
+  render() {
+    const { collection, highlight } = this.props;
+
+    return (
+      <CollectionLink to={collection.url}>
+        <Heading>
+          <IconWrapper>
+            <CollectionIcon collection={collection} />
+          </IconWrapper>
+          <Title text={collection.name} highlight={highlight} />
+          <SecondaryActions>
+            <CollectionMenu collection={collection} position="right" />
+          </SecondaryActions>
+        </Heading>
+      </CollectionLink>
+    );
+  }
+}
+
+const SecondaryActions = styled(Flex)`
+  align-items: center;
+  position: absolute;
+  right: 16px;
+  top: 50%;
+  transform: translateY(-50%);
+`;
+
+const CollectionLink = styled(Link)`
+  display: flex;
+  margin: 10px -8px;
+  padding: 6px 8px;
+  border-radius: 8px;
+  max-height: 50vh;
+  min-width: 100%;
+  max-width: calc(100vw - 40px);
```

Comment:
Can we use `Flex` / flexbox to achieve this layout? It's a very simple lockup and should need so many defined widths etc.

Reply:
My mistake, there was no need to set these properties here, so I'll just remove them.